sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveClientImpl.scala

```diff
@@ -478,13 +478,19 @@ private[hive] class HiveClientImpl(
       ignoredProperties += key -> value
     }
 
-    val excludedTableProperties = HiveStatisticsProperties ++ Set(
+    var excludedTableProperties = HiveStatisticsProperties ++ Set(
       // The property value of "comment" is moved to the dedicated field "comment"
-      "comment",
-      // For EXTERNAL_TABLE, the table properties has a particular field "EXTERNAL". This is added
-      // in the function toHiveTable.
-      "EXTERNAL"
+      "comment"
     )
+    // We should retain the external property if is useful
+    val externalExcluded = properties.get("EXTERNAL").map(_.toLowerCase(Locale.ROOT)) match {
```

Comment:
Hive always show the `EXTERNAL`, but it is not used in normal.

We try best to avoid show the unnecessary property to follow spark since spark we can't modify `EXTERNAL`.